packages/blade/src/_helpers/types.ts

```diff
@@ -83,6 +86,46 @@ type TestID = {
   testID?: string;
 };
 
+/**
+ * Returns the object as it is in web and sets all properties of object to type `never` in react native
+ */
+type MakeObjectWebOnly<T> = Platform.Select<{
+  web: T;
+  native: Record<keyof T, never>;
+}>;
+
+/**
+ * Similar to `Pick` except this returns `never` when value doesn't exist (native `Pick` returns `unknown`).
+ *
+ * You might have to ts-ignore the non-existing type error while using this. This is done so that you can get jsdoc from actual type.
+ *
+ * E.g. This will pick from ViewStyle prop if value exists else returns undefined.
+ *
+ * ```ts
+ * // @ts-expect-error: T passed here may not neccessarily exist. We return `never` type when it doesn't
+ * native: PickIfExist<ViewStyle, T>;
+ * ```
+ */
+type PickIfExist<T, K extends keyof T> = {
+  [P in K]: P extends keyof T ? T[P] : never;
+};
+
+/**
+ * Picks the types based on the platform (web / native).
+ *
+ * E.g.
+ * ```ts
+ * type CSSObject = PickCSSByPlatform<'display'>
+ * // On Web --> This will be all possible web display properties like `block`, `flex`, `inline`, and more.
+ * // On Native --> This will be just `flex` and `none`
+ * ```
+ */
+type PickCSSByPlatform<T extends keyof CSSObject | keyof ViewStyle> = Platform.Select<{
+  web: Pick<CSSObject, T>;
+  // @ts-expect-error: T passed here may not neccessarily exist. We return `never` type when it doesn't
```

Comment:
Any workaround so that we don't have to use `tsexpecterror`. Lord @anuraghazra ?

Reply:
@anuraghazra This is what I am trying to do

https://user-images.githubusercontent.com/30949385/231411096-ba672ad9-c7b7-4291-95d8-8da11d64a3d4.mov

[TS Playground](https://www.typescriptlang.org/play?#code/PQKhAIGEHsAcEsCmATcAzATtAtuARgDYCGyi4IwAsAFCg3ngBCGRAdsgM7hHgAuAnrDLYiAa3isA5nwAWiXEQDGvbl1Y4JRAvQgABDojIzevWBwBcwYNhTwArtgB0inMF0BpRADcJjAOIYhqyIwBx2GD4+UgC0vHKxgogcihjwsLzRiC4c-By88tF4LOwSktFsyAlCsUSSkqXRAGyKaI2IAKwAjI3tAEwAzMg6VNSIAB6w0BgqAkJMxcgAPAAqiQA04AByRDbg4-nsXHmpUgB84AC84KtzAGTgAN7gAPrPRRWvAPzmWztkAL4AbhoNHGk2mfESv148C8iAA8hgAOqIPDMCqXeYVRZsfgbADkrCIMLh+PAAB9wPiAO6o-GnYHUGhEmwcWBKMgABWIvDQU1wD3oewmUxmUO2uyuhOJsMQ+MZ4HAoBAQoYACV4JJjOB1NSNnFhNBkHYCGRAhxoAQ7DDoKxwPAjogZtBwObLXDwAADRy0vCe9DwU0cNaqlXURUMZYyOxcbnEvkYJwAZUQpuUi3OszI1MDBFdTvCdoNXt9-qzoZGirBoshcxTad4i3h6Xgtq4+0Qh0e4F9PzsrFEutYgJ1MrhfYHQ-A-3OV3R7EWqvAzZtrA4AG0JYgALoh8MR-GwHkJ7DRX34oUMmj-EFMu-K8AAWQAmlB4QARACi5BGNAfUYdcAXFZdAsFwfFAiUDIiRJOVwHZRRRFqMgKBoLNwAANSQakkwEU1MUFfdFWVJdIxkQCACkk3faBFHADgZGgE1UHGB0VAkfMoNHWDSNQojwGQB0jyIfgfnxNBTTGMlKUJW05UZa9b3-ciuGApJQJwKk8n4U1KmAyZglYXgODJBCkMkMgfzQqFICTJN4TwAArLIVCuQiIyVMBeOuFTwComi6IYpiCBYsY2PtO1fV4ysI0EtliFEqlCFo0RpKpCTxjS2TgnlJcSP4sjAMA31oltAh+HgrAhGmCqMWQaAklYfEVFYvIIu42V6LwpJoqXSRUmQMSxn4AAvC9qBvO9rLmTl4EQgBJNBPzCvIVg2dxhQOThwFERB+GgNBrgpHa9oOqA7Ic5zlFnR4hXXTl2vcbcfgejsu12-bDuWcBPmue7t3AH5gjhDBGSBW9pq5ObRFspNGH4ONeX5FZNs7baPrO2HLpc46McOrDEBw7qbsRk9HHrFzFncntURe6HFixpyXI2ZYr33GDZTphalpWxsCaJnTEBZtmZwU6h0MYaAxlwwWuCuWbENh+HSeR-E4uE-gsv6+BkHpJSvPDCA-GgLRzB0cBonAJM7FgcEVAQRDSgE+A0DQRBAiM2t1NtZ3Xfdzt7ePflzctvwnT86jaIExA0AkFB2v5mX8IxRmrt4YZbxcNcVDwKWfkl6Xurl27+PVhKxIyqTryAA)

Reply:
In here, the `keyof CSSObject | keyof ViewStyle` generic contraint just becomes `string | number` because CSSObject is a index signature and doesn't specify the concrete CSS object with all the values. 

So right now intellisense either way isn't working for `PickCSSByPlatform` so might be better to do this: 

```ts
type PickCSSByPlatform<T extends string> = Platform.Select<{
  web: Pick<CSSObject, T>;
  native: PickIfExist<ViewStyle, T>;
}>;

type PickIfExist<T, K extends string> = {
  [P in K]: P extends keyof T ? T[P] : never;
};
```

Reply:
But then we loose the JSDoc that comes from actual types

<img width="453" alt="image" src="https://user-images.githubusercontent.com/30949385/231417384-e56561e8-1c29-4aab-ae0b-9fa4bd869828.png">

Expected outcome would be this on react native where it also picks the jsdoc from ViewStyle -
<img width="608" alt="image" src="https://user-images.githubusercontent.com/30949385/231417667-9887969d-7536-469f-8ad3-0d19bd22a845.png">

Reply:
In that case then keep the PickIfExist as is. Adding ts-expect-error should not be an issue.

JSDoc vanishes because in TypeScript there is something called homomorphic types which compiler smartly figures out based on similar keys of the object, If we pass `T extends string` then the compiler isn't able to relate the original object to and just creates a "fresh" object property instead of reusing the original one (which had jsdoc)

```
type PickIfExist<T, K extends keyof T> = {
  [P in K]: P extends keyof T ? T[P] : never;
};
```

Reply:
> Basically just replacing CSSObject to React.CSSProperties. Then the intellisense also works for PickCSSByPlatform.

Yeah in `PickCSSByPlatform` the keep it as React.CSSProperties so that we get intellisese. 

```
type PickCSSByPlatform<T extends keyof ViewStyle | keyof React.CSSProperties> 
```

Reply:
> Yeah in PickCSSByPlatform keep it as React.CSSProperties so that we get intellisese.

yup made this change. thanks!